x/auth/ante/authz_rules_spec.md

```diff
@@ -0,0 +1,65 @@
+# Description
+
+Problem: 
+The existing authorization (authz) module lacks the flexibility to grant permissions (authz grants) for various types of messages along with specific conditions or rules. This limitation constrains users from customizing their transaction behavior based on specific needs or strategies.
+
+## Specific Examples of Limitations:
+Swapping Reward Tokens:
+ - Currently, users cannot set a rule to swap their reward tokens or any other tokens for another token with a specified limit.
```

Comment:
```suggestion
Managing Reward Tokens:
At present, users are able to restake their tokens via authz. But it can do more. Currently users are unable to establish rules for swapping their reward tokens as a strategy as it requires IBCTransfer or PacketForward msgs access. It's not secure to give this grant currently as the recipient address can be anything and grantee can behave maliciously. But if there's a way to restrict recipient address to match with granter's address, this problem is solved. This functionality is necessary to enable users to automate and customize their token management strategies effectively.
```